main.cpp

```diff
@@ -13,8 +16,22 @@ auto program = std::unique_ptr<AstNode>{};
 extern void yylex_destroy();
 
 int main(int argc, char** argv) {
-  /* TODO: read input parameter */
-  output.open("test.ssa");
+  auto options = cxxopts::Options{argv[0], "A simple C compiler."};
+  // clang-format off
+  options.add_options()
+      ("o, output", "Write output to <file>", cxxopts::value<std::string>()->default_value("test.ssa"), "<file>")
+      ("d, dump", "Dump the abstract syntax tree", cxxopts::value<bool>()->default_value("false"))
+      ("h, help", "Display available options")
+      ;
+  // clang-format on
+
+  auto result = options.parse(argc, argv);
+  if (result.count("help")) {
+    std::cerr << options.help() << std::endl;
+    std::exit(0);
+  }
```

Comment:
I think we can add a check for null options and input. In particular, the current code only works when we add `-h` or `--help`, but it would show nothing when we run it without any input. I think we can also show `help()` when there's no input arguments.

```
$  ./vitaminc
```

Reply:
Actually, `./vitaminc` doesn't take any arguments at this moment. It reads input from `stdin`. Therefore, it is valid to have no arguments, which leads `./vitaminc` to hang.

I would recommend deferring this change until we start reading input from a file.